types/types.ts

```diff
@@ -0,0 +1,178 @@
+import { ethers } from 'ethers';
```

Comment:
For my knowledge can you explain again the benefits of adding this file. Is this what gives us granular types per contract?